File: leetcode/cpp/1.two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> hash;
        vector<int> res;
        for(int i=0; i<nums.size(); i++){
            int number = target - nums[i];
            if(hash.find(number) != hash.end()){
                res.push_back(hash[number]);
                res.push_back(i);
                return res;
            }
            hash[nums[i]] = i;
        }
    }
};

File: leetcode/java/1.two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    int[] res = new int[2];
    if (nums == null || nums.length == 0)
      return res;
    Map<Integer, Integer> map = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      int temp = target - nums[i];
      if (map.containsKey(temp)) {
        res[0] = map.get(temp);
        res[1] = i;
        return res;
      } else {
        map.put(nums[i], i);
      }
    }
    return res;
  }
}